docs/modules/ble.md

```diff
@@ -0,0 +1,153 @@
+ble.init(config)
+```
+
+## ble.notify()
+
+This notifies the Bluetooth stack that a new value is available to be read from the characteristic.
+
+#### Syntax
+`ble.notify(notifyvalue)`
+
+#### Parameters
+
+- `notifyvalue` This is the value of the `notify` table entry from the particular characteristic.
+
+#### Returns 
+`nil`
+
+#### Example
+
+```lua
+ble.notify(config.services[1].characteristics[1].notify)
+```
+
+## ble.advertise()
+
+Updates the advertising data field for future advertising frames.
+
+#### Syntax
+`ble.advertise(advertisement)`
+
+#### Parameters
+
+- `advertisement` This string will be placed in future advertising frames as the manufacturer data field. This overrides the a`advertisement` value from the config block.
+
+#### Returns
+`nil`
+
+#### Example
+```lua
+ble.advertise("foo")
+```
+
+## ble.shutdown()
+
+Shuts down the Bluetooth controller and returns it to the state where another `init` ought to work (but currently doesn't). And, at the moment, shutting
+it down doesn't work either -- it appears to corrupt some deep data structures.
+
+#### Syntax
+`ble.shutdown()`
+
+#### Returns
+`nil`
+
+#### Example
+```lua
+ble.shutdown()
+```
+
+## Conventions
+
+## UUID
+
+The service and characteristic identifiers are UUIDs. These are represented in twin-hex. They must be either 4 characters, 8 characters or 32 characters long.
+
+## Configuration Table
+
+The configuration table contains the following keys:
+
+- `name` The name to use to advertise the gadget
+
+- `services` This is a list of tables that define the individual services. The primary service is the first service. Many examples will only have a single service.
+
+- `advertisement` This is a string to be advertised in the mfg data field.
+
+### Service table
+
+The service table contains the following keys:
+
+- `uuid` The UUID of the service. This is a 16 byte string (128 bits) that identifies the particular service. It can also be a two byte string for a well-known service.
+- `characteristics` This is a list of tables, where each entry describes a characateristic (attribute)
+
+### Characteristic table
+
+The characteristic table contains the following keys:
+
+- `uuid` The UUID of the characteristics. This can be either a 16 byte string or a 2 byte string that identifies the particular characteristic. Typically, 2 byte strings are used for well-known characteristics.
+- `type` This is the optional type of the value. It has the same value as a unpack code in the `struct` module.
+- `value` This is the actual value of the characteristic. This will be a string of bytes unless a `type` value is set.
+- `read` This is a function that will be invoked to read the value (and so does not need the `value` entry). It should return a string of bytes (unless `type` is set).
+- `write` This is a function that will be invoked to write the value (and so does not need the `value` entry). It is given a string of bytes (unless `type` is set)
```

Comment:
So if `type` has been set, the argument to write is a number? Maybe clarify this bit slightly, please?

Reply:
I added another paragraph below.